test/e2e/queue.go

```diff
@@ -556,10 +556,17 @@ var _ = Describe("AppWrapper E2E Test", func() {
 		appwrappers = append(appwrappers, aw)
 		err1 := waitAWPodsReady(context, aw)
 		Expect(err1).NotTo(HaveOccurred(), "Expecting pods to be ready for app wrapper: aw-test-jobtimeout-with-comp-1")
-		time.Sleep(90 * time.Second)
-		aw1, err := context.karclient.McadV1beta1().AppWrappers(aw.Namespace).Get(context.ctx, aw.Name, metav1.GetOptions{})
+		var aw1 *arbv1.AppWrapper
+		var err error
+		aw1, err = context.karclient.McadV1beta1().AppWrappers(aw.Namespace).Get(context.ctx, aw.Name, metav1.GetOptions{})
 		Expect(err).NotTo(HaveOccurred(), "Expecting no error when getting app wrapper status")
 		fmt.Fprintf(GinkgoWriter, "[e2e] status of app wrapper: %v.\n", aw1.Status)
+		for aw1.Status.State != arbv1.AppWrapperStateFailed {
```

Comment:
This looks dangerous to me. Tests can be locked here forever. There should be a timer or something to prevent that. Also, this for loop makes line 570 obsolete, as it would always work out of the for loop.

Reply:
@metalcycling each test has a timeout set by the framework, may be we can let framework timer kick-in

Reply:
Mmm... not happy about this but I'll let it pass.